src/app/shared/entry.ts

```diff
@@ -353,10 +343,10 @@ export abstract class Entry implements OnInit, OnDestroy, AfterViewInit {
   }
 
   /**
-   * Deals with redirecting to canonical URL and running discourse call
+   * Deals with redirecting to canonical URL
    * @return {void}
    */
-  redirectAndCallDiscourse(myPage: string): void {
+  redirectToCanonicalURL(myPage: string): void {
```

Comment:
I don't think this method is doing anything now, is it? It calculates some things, but it doesn't do anything with any of the variables. Or am I overlooking something?

Reply:
It updates the path to conform to certain characteristics, though I do think there is code that can be deleted.

Reply:
Oh, I see, it calls `this.decodeURL()`, which changes a property.

Never mind...

Reply:
I do agree we probably need to do some reorganising of this code, but for now I'll leave it as is.